scripts/layerdiff.js

```diff
@@ -0,0 +1,335 @@
+    }
+  }
+
+  if (diffs.centroid || VERBOSE) {
+    const mark = diffs.centroid ? '❌' : '✔️';
+    print(`${mark} Centroid distance:`, Math.round(details.centroid) + 'm');
+    if (VERBOSE > 0) {
+      console.table(details.centroids);
+    }
+  }
+
+  if (diffs.parts || VERBOSE) {
+    const mark = diffs.parts ? '❌' : '✔️';
+    print(`${mark} Geometry parts: left: ${details.parts.left} right: ${details.parts.right}`);
+  }
+
+};
+
+
+const splitFeatures = function (leftJson, rightJson) {
+  let features = [];
+  const notLeft = [];
+
+  if (CHECK_ID) {
+    // Find the elements directly  by the provided ID
+    const id = CHECK_ID;
+    const left = leftJson.features.find(f => getId(f) === id);
+    const right = rightJson.features.find(f => getId(f) === id);
+
+    if (left && right) {
+      features.push({ id, left, right });
+    } else {
+      if (!left) {
+        print('❌ ID not found in the left GeoJSON');
+      }
+      if (!right) {
+        print('❌ ID not found in the right GeoJSON');
+      }
+      yargs.exit(0);
+    }
+  } else {
+    // Join the two arrays merging by the id
+
+    // Put all the left features in a new object
+    features = leftJson.features.map(f => {
+      const id = getId(f);
+      return {
+        id,
+        left: f,
+      };
+    });
+
+    // Put all the right features in the main object
+    // and save the ones that are not found for later
+    rightJson.features.forEach(f => {
+      const id = getId(f);
+      const fIndex = features.findIndex(ff => ff.id === id);
+      if (fIndex === -1) {
+        notLeft.push(f);
+      } else {
+        features[fIndex].right = f;
+      }
+    });
+  }
+
+  // Find if there are any features from the left
+  // without the right side
+  const notRight = features.filter(f => !'right' in f);
+
+  // Work only with features that are on both sides
+  const finalFeatures = features.filter(f => 'right' in f);
+
+  return {
+    notLeft,
+    notRight,
+    features: finalFeatures,
+  };
+};
+
+const pluralize = function (count, noun) {
+  return `${count} ${noun}` + (count > 1 ? 's' : '');
+};
+
+// Execution starts here
+try {
+  // Report given parameters
+  if (VERBOSE) {
+    const fieldId = FIELD_ID ? 'property ' + FIELD_ID : 'default feature id';
+    const checkId = CHECK_ID || 'none';
+    print('================================== Parameters');
+    print(`Identifier: ${fieldId}`);
+    print(`Check a single feature id: ${checkId}`);
+    print(`Area difference: ${AREA_DIFF}`);
+    print(`Centroid distance: ${CENTROID_DIST}`);
+    print(`Check parts: ${CHECK_PARTS}`);
+  }
+
+  const leftPath = argv._[0];
+  const rightPath = argv._[1];
+
+  const reader = new jsts.io.GeoJSONReader();
+
+  // Load GeoJSON files
+  print(`================================== Loading files...`);
+  const leftJson = reader.read(fs.readFileSync(leftPath, 'utf-8'));
+  const rightJson = reader.read(fs.readFileSync(rightPath, 'utf-8'));
```

Comment:
I would also find it useful to be able to pass in a GeoJSON string from stdin, maybe by using `-` as a parameter. Then I could do something like `git show c0b30f:data/france_departments_v7.geo.json | node scripts/layerdiff.js - data/france_departments_v7.geo.json ` to compare a previous revision or different branch to the file on the working branch. 

Something like this, maybe? What do you think @jsanz?
```suggestion
  const leftJson = leftPath === '-'
    ? reader.read(fs.readFileSync('/dev/stdin', 'utf-8'))
    : reader.read(fs.readFileSync(leftPath, 'utf-8'));
  const rightJson = rightPath === '-'
    ? reader.read(fs.readFileSync('/dev/stdin', 'utf-8'))
    : reader.read(fs.readFileSync(rightPath, 'utf-8'));
```

Reply:
thanks! I had that idea but was not sure how to implement it, I changed `/dev/stdin/' by `0` which apparently make it more portable. Change in 21925f1